trollmoves/s3downloader.py

```diff
@@ -0,0 +1,371 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020 Pytroll developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""s3 downloader to be used together with s3stalker from pytroll-collectors.
+S3downloader listens to messages from the s3stalker, and download files to
+configured destination. If the download is successful(ie. file exists on local disk
+a message is published to be used further downstream.
+
+A yaml config file is needed like this:
+---
+logging:
+  log_rotation_days: 1
+  log_rotation_backup: 30
+  logging_mode: DEBUG
+
+subscribe-topic:
+  - /yuhu
+publish-topic: /idnt
+endpoint_url: '<your object store endpoint url'
+access_key: ''
+secret_key: ''
+bucket: <name of the bucket>  # Not needed, else used from the message uri
+download_destination: './'
+
+"""
+
+import os
+import sys
+import yaml
+import queue
+import argparse
+import logging
+from logging import handlers
+from threading import Thread
+from urllib.parse import urlparse
+from posttroll.message import Message
+from posttroll.publisher import Publish
+from posttroll.subscriber import Subscribe
+
+try:
+    from s3fs import S3FileSystem
+except ImportError:
+    S3FileSystem = None
+
+LOGGER = logging.getLogger(__name__)
+# ----------------------------
+# Default settings for logging
+# ----------------------------
+_DEFAULT_TIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+_DEFAULT_LOG_FORMAT = '[%(levelname)s: %(asctime)s : %(name)s] %(message)s'
+
+
+class Listener(Thread):
+
+    def __init__(self, queue, config, subscribe_nameserver):
+        # Thread.__init__(self)
+        super(Listener, self).__init__()
+        self.loop = True
+        self.queue = queue
+        self.config = config
+        self.subscribe_nameserver = subscribe_nameserver
+
+    def stop(self):
+        """Stops the file listener."""
+        LOGGER.debug("Entering stop in FileListener ...")
+        self.loop = False
+        self.queue.put(None)
+
+    def _check_and_put_message_to_queue(self, msg):
+        if not self.loop:
+            LOGGER.warning("Self.loop false in FileListener %s", self.loop)
+            return False
+
+        # Check if it is a relevant message:
+        if self.check_message(msg):
+            LOGGER.info("Put the message on the queue...")
```

Comment:
Is this relevant information? I think this and the next debug messages could be logged when published (see below).